tokio/src/sync/broadcast.rs

```diff
@@ -247,17 +246,14 @@ pub enum TryRecvError {
 /// Data shared between senders and receivers
 struct Shared<T> {
     /// slots in the channel
-    buffer: Box<[Slot<T>]>,
+    buffer: Box<[RwLock<Slot<T>>]>,
 
     /// Mask a position -> index
     mask: usize,
 
     /// Tail of the queue
     tail: Mutex<Tail>,
```

Comment:
This could also probably be a `RwLock` since similar to `Slot`, it's only used for reading values in `recv_ref`.

Reply:
You are probably correct, but it is also not a hot path. I am leaning towards waiting for that change to have a measurable impact, mostly because Mutex is already used here.